agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -49,9 +56,59 @@ public static boolean logSpecialOneTimeFriendlyException(
       //  e.g. wrong host address or cannot reach address due to network issues...
       return false;
     }
+
+    IOException ioException = getCausedByOfType(error, IOException.class);
+    SocketException socketException = getCausedByOfType(error, SocketException.class);
+    if (ioException != null || socketException != null) {
+      if (!alreadySeen.getAndSet(true)) {
+        List<String> missingCiphers = getMissingCiphers(logger);
+        if (missingCiphers.size() == 0) {
+          return false;
+        }
+        logger.error(getCipherFriendlyMessage(url, missingCiphers));
+      }
+      return true;
+    }
     return false;
   }
 
+  private static List<String> getMissingCiphers(Logger logger) {
+    final List<String> missingCiphers = new ArrayList<>();
+    final List<String> expectedCiphers =
+        Arrays.asList(
+            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+            "TLS_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_RSA_WITH_AES_256_CBC_SHA256",
+            "TLS_RSA_WITH_AES_128_CBC_SHA256",
+            "TLS_RSA_WITH_AES_256_CBC_SHA",
+            "TLS_RSA_WITH_AES_128_CBC_SHA");
```

Comment:
this is not the correct list for most of the endpoints, e.g.
https://www.ssllabs.com/ssltest/analyze.html?d=westus2-2.in.applicationinsights.azure.com